src/sources

```diff
@@ -271,6 +274,24 @@ $(SRC)/checknest.sml
 $(SRC)/compiler.sig
 $(SRC)/compiler.sml
 
+$(SRC)/getinfo.sig
+$(SRC)/getinfo.sml
+
+$(SRC)/json.sig
+$(SRC)/json.sml
+
+$(SRC)/fromjson.sig
+$(SRC)/fromjson.sml
+
+$(SRC)/lspspec.sml
+
+$(SML_LIB)/basis/mlton.mlb
```

Comment:
This MLton-specific line shouldn't be included in this file.  Instead, it probably belongs in `prefix.mlb`, which the Makefile uses to create the input to MLton.  Ur/Web is also supposed to build with SML/NJ, which is why I put most of the build-configuration content in `sources` but keep some of it in compiler-specific files.

Reply:
And, actually, if one of your files only builds with MLton, we should think harder about the way to support it.

Reply:
Yeah, I've included the MLton Thread module so elaboration can go on in the background while in the foreground the server can stay responsive to requests. This is a huge usability improvement, but it does indeed complicate the build story. Just thinking out loud: if the Thread module isn't there, it could still function, just with every operation blocking the other. In a development situation, that's not really a problem. I'm unfamiliar with how to set this up. I suppose I should make Thread module with dummy function declarations that would be used if you run the compiler with SML/NJ?

Reply:
Yeah, that sounds like a good idea!

Reply:
Ok, Done. I've "mocked" the BgThread module though, I hope that's OK. Mocking the actual Mlton Thread, Signal, Timer, etc modules is much much harder and more work so I think it's OK.

Reply:
Yeah, it makes sense to design a new module signature exposing just what is needed here.